### PR TITLE
Reduce dmesg verbosity2

### PR DESCRIPTION
**Pull request by plbossart**
Two more cases of mildly useful info messages.

### CODE REVIEW

File: sound/soc/soc-pcm.c [hunk 1]
```diff
@@ -2963,9 +2963,9 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 
 	pcm->no_device_suspend = true;
 out:
-	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
-		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
-		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
+	dev_dbg(rtd->card->dev, "%s <-> %s mapping ok\n",
+		(rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
+		(rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
```
Comment (bardliao) [inline sound/soc/soc-pcm.c:2969]: This one is helpful for me. I can see which rtd is mapping ok with dynamic debug disabled. So I will know which rtd may have problem before dynamic debug is enabled.

Comment (plbossart) [reply sound/soc/soc-pcm.c:2969]: @bardliao I have personally never used this information, I don't even know what it means. Can you clarify what sort of error classes can be identified with this trace?

Comment (plbossart) [reply sound/soc/soc-pcm.c:2969]: @bardliao If you look at the code in soc_pcm_new, what you really want is a better message when the following code fails. Having a trace when everything works fine is not really useful.

````c
	ret = snd_soc_pcm_component_new(rtd);
	if (ret < 0) {
		dev_err(rtd->dev, "ASoC: pcm constructor failed: %d\n", ret); <<< HERE
		return ret;
	}

	pcm->no_device_suspend = true;
out:
	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
	return ret;
````

**Comment by plbossart** [comment]
update with better error messages.

File: sound/soc/soc-pcm.c [hunk 2]
```diff
@@ -2891,8 +2891,8 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 			capture, &pcm);
 	}
 	if (ret < 0) {
-		dev_err(rtd->card->dev, "ASoC: can't create pcm for %s\n",
-			rtd->dai_link->name);
+		dev_err(rtd->card->dev, "ASoC: can't create pcm for %s: %d\n",
+			rtd->dai_link->name, ret);
```
Comment (kv2019i) [inline sound/soc/soc-pcm.c]: Hmm, would it be useful to print out "new_name" as well? It seems it's filled in all possible cases and probably gives a good hint at what goes wrong.

File: sound/soc/soc-pcm.c [hunk 3]
```diff
@@ -2891,8 +2891,8 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
...
 		return ret;
 	}
 	dev_dbg(rtd->card->dev, "ASoC: registered pcm #%d %s\n",num, new_name);
```
Comment (kv2019i) [inline sound/soc/soc-pcm.c:2898]: It's used in the dev_dbg() print here after all...

File: sound/soc/soc-pcm.c [hunk 4]
```diff
@@ -2957,7 +2957,8 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 
 	ret = snd_soc_pcm_component_new(rtd);
 	if (ret < 0) {
-		dev_err(rtd->dev, "ASoC: pcm constructor failed: %d\n", ret);
+		dev_err(rtd->dev, "ASoC: pcm constructor failed for %s: %d\n",
+			rtd->dai_link->name, ret);
```
Comment (kv2019i) [inline sound/soc/soc-pcm.c]: Ditto above

File: sound/soc/soc-pcm.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/soc-pcm.c]: good suggestion @kv2019, added new_name in update

**Comment by plbossart** [comment]
update with even better messages suggested by @kv2019i

File: sound/soc/soc-pcm.c (hunk 1 above)
Comment (bardliao) [reply sound/soc/soc-pcm.c:2969]: > @bardliao If you look at the code in soc_pcm_new, what you really want is a better message when the following code fails. Having a trace when everything works fine is not really useful.
> 
> ```c
> 	ret = snd_soc_pcm_component_new(rtd);
> 	if (ret < 0) {
> 		dev_err(rtd->dev, "ASoC: pcm constructor failed: %d\n", ret); <<< HERE
> 		return ret;
> 	}
> 
> 	pcm->no_device_suspend = true;
> out:
> 	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
> 		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
> 		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
> 	return ret;
> ```

Thanks @plbossart You are right.

**Comment by kv2019i** [comment]
Checkpatch warning from existing code, not added by this PR, so looks good to go.

**Pull request merged**